lib/std/thread.zig

```diff
@@ -484,6 +484,9 @@ pub const Thread = struct {
             };
             return @intCast(usize, count);
         }
+        if (std.Target.current.os.tag == .haiku) {
+            return 1;        // TODO:  implement me
```

Comment:
Please remove all bugs such as this from this pull request. You can replace this with 

```suggestion
            @panic("TODO implement std.Thread.cpuCount for Haiku");
```

Or better yet:

```suggestion
            @compileError("TODO implement std.Thread.cpuCount for Haiku");
```

No wrong implementations please!

Reply:
makes sense, and have placed in the suggestions in (hopefully) the appropriate places. might have implemented some parts in other areas which hopefully also addresses the problematic code spots